### ISSUE TITLE
Unexpected classifier v2 classificaiton annotation format(s) for dropdown tasks

### ISSUE THREAD

**Issue by camallen** [zooniverse/front-end-monorepo#2119]
I'm investigating a broken classification data export in the API and have found 4 classifications (all by the same user) submitted by the FEM classifier (`classifier_version: '2.0'`) but that are for `taskType: 'dropdown'` not `dropdown-simple` (see the sample data below)

Based on my reading of the FEM code for registered task types https://github.com/zooniverse/front-end-monorepo/blob/813f0334eb9543a9163ac5c312c7654722f3458f/packages/lib-classifier/src/plugins/tasks/index.js#L13-L25
and the simple dropdown ADR https://github.com/zooniverse/front-end-monorepo/blob/master/docs/arch/adr-30.md#consequences, it seems that FEM annotations should not have the `taskType: 'dropdown'` value as that isn't a registered tasks type in FEM land, is this correct?

I believe the main issue at play here is these problematic classifications annotations are from the v2 classifier and have a mixture of PFE and FEM annotation formats, specifically I have these questions:

1. Are these valid FEM annotations? 
2. Can the v2 (FEM) classifier submit annotations for dropdowns that have more than 1 select? 
    + If so then I need guidance to figure out which select the submitted value aligns to.
3. Am I incorrectly interpreting the annotation format from the v2 classifier?

I found this issue when working on a classifier v2 annotation format exporter in panoptes, which handles the different annotation for `dropdown-simple` tasks which is working fine, https://github.com/zooniverse/panoptes/pull/3593. Once we figure out what's up with these annotations I can proceed to fix the API classification data exporters for the FEM v2 classification annotations formats.

Here is a sample of the classification data that is causing issues:
```
#<Classification:
 id: 320640562,
 project_id: 11429,
 user_id: redacted,
 workflow_id: 18118,
 annotations:
  [{"task"=>"T1", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T2", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T3", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T4", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T5", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T7", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T8", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T9", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T10", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T11", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T4", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T5", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T7", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T8", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T9", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T10", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T11", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T13", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T14", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T15", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T16", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T17", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T19", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T20", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T21", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T22", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T23", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T25", "value"=>nil, "taskType"=>"dropdown"}, # this is the problem task annotation right here...
   {"task"=>"T26", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T27", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T28", "value"=>nil, "taskType"=>"dropdown-simple"},
   {"task"=>"T29", "value"=>nil, "taskType"=>"dropdown-simple"}],
 created_at: Tue, 30 Mar 2021 16:49:23 UTC +00:00,
 updated_at: Tue, 30 Mar 2021 16:49:23 UTC +00:00,
 workflow_version: "2.11"
 metadata: 
    {"classifier_version"=>"2.0"} # I redacted the rest
```
and the associated workflow task below (this is the 2.11 version of the workflow as used when exporting the annotation above)
```
# problematic workflow Task - note 2 selects (dropdowns)
# in my understanding having more than 1 select should break the wiring of the classifier and no classification should be submitted
{"help"=>"T25.help",
 "type"=>"dropdown",
 "selects"=>
  [{"id"=>"cd1c1c2ffc297",
    "title"=>"Select Circumstance #21",
    "options"=>
     {"*"=>
       [{"label"=>"T25.selects.0.options.*.0.label", "value"=>"2db17b653c0cf"},
        {"label"=>"T25.selects.0.options.*.1.label", "value"=>"58d7d68f3a22c"},
        {"label"=>"T25.selects.0.options.*.2.label", "value"=>"f9368bbaa19c7"},
        {"label"=>"T25.selects.0.options.*.3.label", "value"=>"243905a9b1b51"},
        {"label"=>"T25.selects.0.options.*.4.label", "value"=>"2d1b73e02926f"},
        {"label"=>"T25.selects.0.options.*.5.label", "value"=>"ed34cc05001ca"},
        {"label"=>"T25.selects.0.options.*.6.label", "value"=>"d96161a08626e"},
        {"label"=>"T25.selects.0.options.*.7.label", "value"=>"b4af1681aafd9"},
        {"label"=>"T25.selects.0.options.*.8.label", "value"=>"d78f16c4d8ad"},
        {"label"=>"T25.selects.0.options.*.9.label", "value"=>"d39597d25ca0f"},
        {"label"=>"T25.selects.0.options.*.10.label", "value"=>"75753ce2df7f9"},
        {"label"=>"T25.selects.0.options.*.11.label", "value"=>"fed66fdd3fcdd"},
        {"label"=>"T25.selects.0.options.*.12.label", "value"=>"2976671f2d3e8"},
        {"label"=>"T25.selects.0.options.*.13.label", "value"=>"d9b71212f6b3b"},
        {"label"=>"T25.selects.0.options.*.14.label", "value"=>"53e7641fc823d"},
        {"label"=>"T25.selects.0.options.*.15.label", "value"=>"17474857fd73f"},
        {"label"=>"T25.selects.0.options.*.16.label", "value"=>"7097b605c85e1"},
        {"label"=>"T25.selects.0.options.*.17.label", "value"=>"63a98408f2935"},
        {"label"=>"T25.selects.0.options.*.18.label", "value"=>"9d628f1cb9692"}]},
    "required"=>true,
    "allowCreate"=>false},
   {"id"=>"605671ee9452f", "title"=>"Other (Please transcribe)", "options"=>{}, "required"=>false, "condition"=>"cd1c1c2ffc297", "allowCreate"=>true}],
 "instruction"=>"T25.instruction"}
```

**Comment by eatyourgreens** [comment]
@camallen could you post the workflow task for T23 or T26, so that we can compare with a working task?

**Comment by eatyourgreens** [comment]
I've got a test workflow set up on staging (project 908, workflow 3489) with a dropdown task that has two menus. It throws errors in the console, as expected.
```
T0 dropdown is not a supported task type
```
and later
```
at path "/type" value `"dropdown"` is not assignable to type: `"dropdown-simple"`
```

**Comment by eatyourgreens** [comment]
Here's `task.selects` for my test. I think it the same as the broken task above.
```json
[
  {
    "id": "c0a8efe878208",
    "title": "Main Dropdown",
    "options": {
      "*": [
        {
          "label": "one",
          "value": "3a5e7c30a1f81"
        },
        {
          "label": "two",
          "value": "ecc0e62c6c11"
        },
        {
          "label": "three",
          "value": "57ca0302a9e4d"
        },
        {
          "label": "four",
          "value": "f8a9fcd7c6cc6"
        }
      ]
    },
    "required": true,
    "allowCreate": false
  },
  {
    "id": "1de4d3316327f",
    "title": "text entry",
    "options": {},
    "required": false,
    "condition": "c0a8efe878208",
    "allowCreate": true
  }
]
```

**Comment by srallen** [comment]
@camallen 

> Are these valid FEM annotations?

No, likely produced during development as a means to not have to do any lab work by adding a backwards compatibility for `dropdown` task to be loaded as `dropdown-simple`. See: https://github.com/zooniverse/front-end-monorepo/tree/master/packages/lib-classifier/src/plugins/tasks/SimpleDropdownTask/models/helpers/legacyDropdownAdapter

I think we should remove this adapter.

> Can the v2 (FEM) classifier submit annotations for dropdowns that have more than 1 select?

No, the simple dropdown only supports one select list up to 20 values and only a single value can be chosen. Think of it like a single choice task, but with a different UI.

**Comment by camallen** [comment]
> @camallen could you post the workflow task for T23 or T26, so that we can compare with a working task?
workflow_id: 18118
major_number: 2
minor_number: 11
Task T23
```
workflow_information.at_version.tasks['T23']
=> {"help"=>"T23.help",
 "type"=>"dropdown",
 "selects"=>
  [{"id"=>"8c3ed79fe2779",
    "title"=>"Select Circumstance #20",
    "options"=>
     {"*"=>
       [{"label"=>"T23.selects.0.options.*.0.label", "value"=>"fbb3acca5d6b1"},
        {"label"=>"T23.selects.0.options.*.1.label", "value"=>"d0f84b54d2ea1"},
        {"label"=>"T23.selects.0.options.*.2.label", "value"=>"a78935c293e2f"},
        {"label"=>"T23.selects.0.options.*.3.label", "value"=>"b30b4451671e3"},
        {"label"=>"T23.selects.0.options.*.4.label", "value"=>"5cf41290ecd2d"},
        {"label"=>"T23.selects.0.options.*.5.label", "value"=>"4f904f40dedcd"},
        {"label"=>"T23.selects.0.options.*.6.label", "value"=>"2ca9ff5f6d235"},
        {"label"=>"T23.selects.0.options.*.7.label", "value"=>"639cc6e1f1252"},
        {"label"=>"T23.selects.0.options.*.8.label", "value"=>"11c8147b89178"},
        {"label"=>"T23.selects.0.options.*.9.label", "value"=>"405cdb781a22a"},
        {"label"=>"T23.selects.0.options.*.10.label", "value"=>"3f681aa73a11d"},
        {"label"=>"T23.selects.0.options.*.11.label", "value"=>"7dfb36059f909"},
        {"label"=>"T23.selects.0.options.*.12.label", "value"=>"ebbe5b11edde1"},
        {"label"=>"T23.selects.0.options.*.13.label", "value"=>"01b616864a36e"},
        {"label"=>"T23.selects.0.options.*.14.label", "value"=>"97f88c3416071"},
        {"label"=>"T23.selects.0.options.*.15.label", "value"=>"7edaa6052f714"},
        {"label"=>"T23.selects.0.options.*.16.label", "value"=>"01cd86304c445"},
        {"label"=>"T23.selects.0.options.*.17.label", "value"=>"54c3bdb63ab12"},
        {"label"=>"T23.selects.0.options.*.18.label", "value"=>"d025190a56103"},
        {"label"=>"T23.selects.0.options.*.19.label", "value"=>"4f0a2b97d99ad"}]},
    "required"=>true,
    "allowCreate"=>false}],
 "instruction"=>"T23.instruction"}
```
workflow_id: 18118
major_number: 2
minor_number: 11
Task T26
```
workflow_information.at_version.tasks['T26']
=> {"help"=>"T26.help",
 "type"=>"dropdown",
 "selects"=>
  [{"id"=>"8499f4d49c1ef",
    "title"=>"Select Circumstance #22",
    "options"=>
     {"*"=>
       [{"label"=>"T26.selects.0.options.*.0.label", "value"=>"6d024d279272e"},
        {"label"=>"T26.selects.0.options.*.1.label", "value"=>"c0405a4a3fbc8"},
        {"label"=>"T26.selects.0.options.*.2.label", "value"=>"089de575d516"},
        {"label"=>"T26.selects.0.options.*.3.label", "value"=>"fd19b455e2811"},
        {"label"=>"T26.selects.0.options.*.4.label", "value"=>"36c961869a94b"},
        {"label"=>"T26.selects.0.options.*.5.label", "value"=>"73651b716b568"},
        {"label"=>"T26.selects.0.options.*.6.label", "value"=>"3ff220ec44c02"},
        {"label"=>"T26.selects.0.options.*.7.label", "value"=>"c2a8150e089a6"},
        {"label"=>"T26.selects.0.options.*.8.label", "value"=>"a3b59824f19d1"},
        {"label"=>"T26.selects.0.options.*.9.label", "value"=>"73c16318e6ccd"},
        {"label"=>"T26.selects.0.options.*.10.label", "value"=>"848df3917d2cc"},
        {"label"=>"T26.selects.0.options.*.11.label", "value"=>"94e1e6c75a1b9"},
        {"label"=>"T26.selects.0.options.*.12.label", "value"=>"dbd8d77d74908"},
        {"label"=>"T26.selects.0.options.*.13.label", "value"=>"a829ab708b6c6"},
        {"label"=>"T26.selects.0.options.*.14.label", "value"=>"7b9818f80864f"},
        {"label"=>"T26.selects.0.options.*.15.label", "value"=>"b76363175661d"},
        {"label"=>"T26.selects.0.options.*.16.label", "value"=>"8142f81b0f483"},
        {"label"=>"T26.selects.0.options.*.17.label", "value"=>"514c684ea753b"},
        {"label"=>"T26.selects.0.options.*.18.label", "value"=>"44706928fd22d"},
        {"label"=>"T26.selects.0.options.*.19.label", "value"=>"01d14429da9f8"}]},
    "required"=>true,
    "allowCreate"=>false}],
 "instruction"=>"T26.instruction"}
```

**Comment by camallen** [comment]
> No, likely produced during development as a means to not have to do any lab work by adding a backwards compatibility for `dropdown` task to be loaded as `dropdown-simple`. See: https://github.com/zooniverse/front-end-monorepo/tree/master/packages/lib-classifier/src/plugins/tasks/SimpleDropdownTask/models/helpers/legacyDropdownAdapter

The classifications are for user `msalmon` apparently not from a development machine so this would have been a deployed (staging possibly) version of the code
```
classification.user.login
=> "msalmon"
```

> > Can the v2 (FEM) classifier submit annotations for dropdowns that have more than 1 select?
> 
> No, the simple dropdown only supports one select list up to 20 values and only a single value can be chosen. Think of it like a single choice task, but with a different UI.

Then this is what is worry me, i've got an annotation that doesn't record the select it was an answer for** and may be a bug that will impact on downstream data if we launch this tools. 

I think we need to figure out how the FEM can create annotations like this (it may be linked to the adapter but maybe not) before we can safely proceed. It would have been nice to have the classification metadata record the version of client code so we could checkout a specific revision of the code for testing, that might be a useful feature to add in.

If we decide that the above is not something we can reproduce then i'm happy enough to modify / delete the erroneous classification but it would be terrible to launch a project with these tools and then retrospectively have to ignore or reconstruct data. Ideally we use this as a warning to avoid launching projects with classifier tools that may have data veracity issues,

** the annotations values used to be in array format and the index used to align to the index of the selects in the task so we knew which value and task went together.

problematic classifications id's and created at
|  id | created at (server received time) |
| --------------- | --------------- |
| 320640562 | 2021-03-30 16:49:23 UTC |
| 320640678 | 2021-03-30 16:50:04 UTC |
| 320640948 | 2021-03-30 16:51:18 UTC |
| 320641025 | 2021-03-30 16:51:38 UTC |

**Comment by eatyourgreens** [comment]
I think Martin Salmon has both the staging and production URLs for HMS NHS, so the code version he was using would depend on which URL he was using to test out his workflow. On March 30th, the production release was 9e6f7848448e. There's then 3 PRs after that were released to staging on 30th March.

**Comment by srallen** [comment]
> Then this is what is worry me, i've got an annotation that doesn't record the select it was an answer for** and may be a bug that will impact on downstream data if we launch this tools.

> I think we need to figure out how the FEM can create annotations like this (it may be linked to the adapter but maybe not) before we can safely proceed. It would have been nice to have the classification metadata record the version of client code so we could checkout a specific revision of the code for testing, that might be a useful feature to add in.

I completely agree. We should be very cautious here. I suspect that a version of this was shared with an EC team member and that this is an artifact of the development process, but likely could still be a bugged adapter or task code.

We have an old idea of validating payloads: https://github.com/zooniverse/front-end-monorepo/issues/805
As well as including a schema definition or a link to one with the classification: https://github.com/zooniverse/front-end-monorepo/blob/master/docs/arch/adr-07.md

Both ideas still unimplemented. We could include the git SHA as part of the classification metadata?

**Comment by camallen** [comment]
> We could include the git SHA as part of the classification metadata?

You could include it like the `classifier_version: ` metadata key was added and only added for FEM classifications, e.g. `revision: 9e6f784` ?

**Comment by srallen** [comment]
> You could include it like the classifier_version: metadata key was added and only added for FEM classifications, e.g. revision: 9e6f784 ?

Yep, I think it would be fairly simple to add and something we can do quickly. This won't help debugging this, but useful for the future especially if we're trying to correlate a classification to a Sentry error report where we're also using the git SHA for the release version.

**Comment by eatyourgreens** [comment]
I've set up a test workflow in the project builder,  with two dropdowns: one valid and one invalid. Classifying on that workflow (in demo mode) produces valid classifications, though.
https://localhost:8080/?demo=true&project=908&workflow=3489

I've also copied that workflow setup into a spec, in 3d1cf3975a9e66ef81269baf75b55b43b41aefde It sets up the workflow, then checks that there's only task, and only one `dropdown-simple` annotation.

**Comment by eatyourgreens** [comment]
I've updated those tests to also check the serialised classification that's submitted to Panoptes. To run them:
```
git checkout test-invalid-dropdown-workflows
cd packages/lib-classifier
yarn test
```

The output should look something like this:
```
yarn run v1.22.10
$ NODE_PATH=. NODE_ENV=test mocha --config ./test/.mocharc.json "./{src,test}/**/*.spec.js"
Fetching more subjects


  workflow with invalid dropdown task
T0 dropdown is not a supported task type
Error: [mobx-state-tree] Error while converting `{"help":"","type":"dropdown","selects":[{"id":"c0a8efe878208","title":"Main Dropdown","options":{"*":[{"label":"one","value":"3a5e7c30a1f81"},{"label":"two","value":"ecc0e62c6c11"},{"label":"three","value":"57ca0302a9e4d"},{"label":"four","value":"f8a9fcd7c6cc6"}]},"requ......":"T0"}` to `(DataVisAnnotationTask | DrawingTask | MultipleChoiceTask | SimpleDropdownTask | SingleChoiceTask | SubjectGroupComparisonTask | SurveyTask | TextTask | TranscriptionTask)`:

    at path "/type" value `"dropdown"` is not assignable to type: `"dropdown-simple"` (Value is not a literal "dropdown-simple").
    at path "/options" value `undefined` is not assignable to type: `MenuOptions` (Value is not an array).
    at fail$1 (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx-state-tree/dist/mobx-state-tree.js:3132:12)
    at typecheck (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx-state-tree/dist/mobx-state-tree.js:2783:15)
    at typecheckInternal (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx-state-tree/dist/mobx-state-tree.js:2769:9)
    at valueAsNode (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx-state-tree/dist/mobx-state-tree.js:4570:5)
    at reconcileArrayChildren (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx-state-tree/dist/mobx-state-tree.js:4541:27)
    at ArrayType.willChange (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx-state-tree/dist/mobx-state-tree.js:4384:38)
    at interceptChange (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:2993:37)
    at ObservableArrayAdministration.spliceWithArray (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:3162:26)
    at Proxy.push (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:3303:13)
    at /Users/odonnell/zooniverse/front-end-monorepo/packages/lib-classifier/src/store/WorkflowStepStore/WorkflowStepStore.js:147:24
    at /Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:3428:29
    at Array.forEach (<anonymous>)
    at Proxy.forEach (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:3427:40)
    at /Users/odonnell/zooniverse/front-end-monorepo/packages/lib-classifier/src/store/WorkflowStepStore/WorkflowStepStore.js:141:23
    at MSTMap.forEach (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:3654:26)
    at Object.setTasks (/Users/odonnell/zooniverse/front-end-monorepo/packages/lib-classifier/src/store/WorkflowStepStore/WorkflowStepStore.js:140:18)
    at executeAction (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:927:19)
    at bound setTasks (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:914:16)
    at runMiddleWares (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx-state-tree/dist/mobx-state-tree.js:2597:40)
    at runWithActionContext (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx-state-tree/dist/mobx-state-tree.js:2460:16)
    at Object.setTasks (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx-state-tree/dist/mobx-state-tree.js:2487:16)
    at setStepsAndTasks (/Users/odonnell/zooniverse/front-end-monorepo/packages/lib-classifier/src/store/WorkflowStepStore/WorkflowStepStore.js:129:12)
    at workflowDisposer.name (/Users/odonnell/zooniverse/front-end-monorepo/packages/lib-classifier/src/store/WorkflowStepStore/WorkflowStepStore.js:88:11)
    at reactionRunner (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:2106:9)
    at trackDerivedFunction (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:776:24)
    at Reaction.track (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:1806:22)
    at Reaction.onInvalidate (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:2087:18)
    at Reaction.runReaction (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:1772:26)
    at runReactionsHelper (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:1909:35)
    at reactionScheduler (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:1887:47)
    at /Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:1916:71
    at batchedUpdates$1 (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/react-dom/cjs/react-dom.development.js:21856:12)
    at reactionScheduler (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:1916:47)
    at runReactions (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:1892:5)
    at endBatch (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:1592:9)
    at _endAction (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:980:5)
    at executeAction (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:934:9)
    at ModelType.create (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mobx/lib/mobx.js:914:16)
    at mockStore (/Users/odonnell/zooniverse/front-end-monorepo/packages/lib-classifier/test/mockStore/mockStore.js:36:31)
    at Context.<anonymous> (/Users/odonnell/zooniverse/front-end-monorepo/packages/lib-classifier/src/store/index.spec.js:97:13)
    at callFn (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mocha/lib/runnable.js:366:21)
    at Hook.Runnable.run (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mocha/lib/runnable.js:354:5)
    at next (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mocha/lib/runner.js:510:10)
    at Immediate._onImmediate (/Users/odonnell/zooniverse/front-end-monorepo/node_modules/mocha/lib/runner.js:572:5)
    at processImmediate (internal/timers.js:461:21)
Fetching more subjects
    the workflow
      ✓ should have one step
      ✓ should have 3 tasks
      ✓ should have 2 dropdown tasks
    workflow steps
      ✓ should have one step
      ✓ should have one dropdown simple task
    the active classification
      ✓ should have one annotation
      ✓ should have a dropdown-simple annotation for task T1
      ✓ should not have a dropdown annotation for task T0
    submitted classifications
      ✓ should have one annotation
      ✓ should have a dropdown-simple annotation for task T1
      ✓ should not have a dropdown annotation for task T0


  11 passing (142ms)

✨  Done in 27.24s.
```

**Comment by eatyourgreens** [comment]
`annotation.taskType` is defined as `types.string`, so could take any string value eg. a dropdown annotation could have `taskType: 'drawing'` without erroring. I can't see any code paths that would lead to that happening in the classifier, but I've opened #2122 to tighten up the type definitions anyway.

**Comment by eatyourgreens** [comment]
I've opened #2123 to add the latest git commit to the classification, as `classification.metadata.revision`.

**Comment by camallen** [comment]
Based on the feedback from @eatyourgreens about the current error states with the same task wiring, then if we can't reproduce or isolate the error behaviour then I'm inclined to think that these were 4 errant classifications resulting from a problematic deploy that we've since fixed. 

With the extra addition PRs to tighten up the behaviours on the v2 task annotation formats then to fix this EC project classification data exports I suggest moving the 4 errant classifications to a test project / workflow. 

FWIW - I still this is very important and something we need to keep an eye on. However failing to create a reproduction with all the details or finding more problematic classification data means we can't be sure what's happening. 

Let me know if you agree with the above and i can get to work cleaning up the data and fixing the exports.

**Comment by srallen** [comment]
@camallen that sounds reasonable to me and we can have a close look at the related projects who are beta testing the feature as it is being tested more broadly. I'd like us to consider some of the schema validations that we've proposed in the past for near term future work as well to continue to tighten up the classifier's behavior.

**Comment by eatyourgreens** [comment]
The other project to keep an eye on would be `12268` (Scarlets & Blues.) I think they also have dropdown menus.

**Comment by camallen** [comment]
OK - I've rehoused these 4 classifications on one of my own projects, https://www.zooniverse.org/lab/16034/workflows/18451

Code used to rehome them:
``` ruby
borked_cls = Classification.where(id: [320640562, 320640678, 320640948, 320641025])
# double check them 
borked_cls.count
borked_cls.map(&:workflow_id)
borked_cls.map(&:project_id)
# now move them
borked_cls.update_all(project_id: 16034, workflow_id: 18451)
```
